aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,194 @@
+
+#### ASP.NET Core Blazor Server
+
+ASP.NET Core Blazor Server has the advantage of direct access to server resources, while providing model for building rich, interactive, and composable user interfaces in C#. Razor components in Blazor Server apps are UI elements such as a page, dialog, or data entry form, that handle client-side web UI interactions. The client-side web UI interactions are handled over a websocket connection and can span multiple connections to increase speed. Razor components keep track of the state of the client-side DOM and efficiently apply updates to only the components that need it.
+
+Blazor Server benefits:
+
+* Share app logic across server and client.
+* Flexibility of direct access to server resources. For example:
+  * Database.
+  * Secrets, such as access key-values used for API calls to [Azure Storage](/azure/storage/).
+* Uses a token-based security model, rather than using cookies. For example, cookies may not be enabled on the browser.
+
+To get started with ASP.NET Core Blazor Server, see [Get started with Blazor](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). For an overview of ASP.NET Core Blazor Server, its architecture and benefits, see <xref:blazor/index>.
+
+### ASP.NET Core MVC plus hosted Blazor WebAssembly
+
+MVC and Blazor are both part of the ASP.NET Core framework and are designed to be used together. Razor components can be integrated into Razor Pages and MVC apps in a hosted Blazor WebAssembly solution. When the page or view is rendered, components can be prerendered at the same time.
+
+Choose ASP.NET Core MVC plus Blazor WebAssembly when these benefits best fit your web UI app needs:
+
+* Prerendering:
+  * Executes Razor components on the server and renders them into a page or view.
+      * Improves the perceived load time of the app while setting up interactivity.
+* Add islands of interactivity to existing views (pages) with the [Component Tag Helper](xref:mvc/views/tag-helpers/builtin-th/component-tag-helper).
+
+To get started with ASP.NET Core MVC plus Blazor WebAssembly, see <xref:blazor/components/prerendering-and-integration>.
+
+### ASP.NET Core Single Page Application (SPA) with JavaScript Frameworks such as Angular and React
+
+Build client-side logic for ASP.NET Core apps using popular JavaScript frameworks, like [Angular](https://angular.io/) or [React](https://facebook.github.io/react/). ASP.NET Core provides project templates for Angular and React.
+
+Benefits of ASP.NET Core SPA with JavaScript Frameworks, in addition to the client rendering benefits previously listed:
+
+* The JavaScript runtime environment is already provided with the browser.
+* Large community and mature ecosystem.
+* Build client-side logic for ASP.NET Core apps using popular JS frameworks, like Angular and React, or build your client-side logic with .NET using Blazor.
```

Comment:
This last bullet doesn't really sound like a benefit of building JS based SPAs with ASP.NET Core.